### ISSUE TITLE
Embed mp4 videos from Imgur

### ISSUE THREAD

**Issue by nolanw** [Awful/Awful.app#861]
> Is there any support for embedded mp4's from imgur in posts? I installed the app specifically because these common mp4 videos wouldn't embed on iOS Chrome like they do on Windows Chrome, but it was a vain attempt, the Awful app still doesn't embed them (although they do show up as links, which is a step up from iOS Chrome which shows them as non-interactive still images)
> — [Dumb Lowtax](https://forums.somethingawful.com/showthread.php?noseen=0&threadid=3837546&perpage=40&pagenumber=42#post492447723)

**Comment by nolanw** [comment]
> https://forums.somethingawful.com/showthread.php?threadid=3553037&perpage=40&pagenumber=589#post492619354
> 
> I have achieved a heckin sorcery:
> 
> If you img tag an MP4, it will only play in the app. If you url tag an MP4, it will only play on Desktop.
> 
> So. I have tagged one as img, and the other as url.
> 
> Guess what?
> 
> Video appears ONLY ONCE, despite. So browse on mobile! Browse on the web! See the difference.
> 
> Can this process be made easier?
> — [Wise Fwom Yo Gwave](https://forums.somethingawful.com/showthread.php?noseen=0&threadid=3837546&perpage=40&pagenumber=42#post492619618)

---

> Actually, on desktop Safari, I can see both. What happens is the img tagged one (the second one, right?) gets served up by imgur as a gifv, which safari (and awful, and probably chrome? I assume this is a webkit thing) understands. That happily loops forever, silently.
> 
> The first video, which is url tagged, embeds as an mp4, and won't do that. (although I'm pretty sure I set a flag somewhere to prevent autoplaying videos. If I click through to show controls, I can play the video, sound and all.)
> 
> I only checked with firefox, but the img-tagged gifv version didn't work in firefox, it just displays a broken file placeholder.
> 
> Out of curiosity, how did you upload your video? When I tried I went through the pain in the ass imgur.com/upload page (and used its share link, which is how I got the img-tagged gifv in the first place).
> 
> I'd be curious to know if there's a proper solution to this, though. To be honest, I just used the img tags because I couldn't remember which ones worked for embedding a video like that. I kinda feel like once upon a time imgur just served a suitable format to whatever browser and it just worked, but it's not like SA's bbcode listing has ever been changed, so I'm not sure what's the "correct" way to post gifv/webm/whatevers.
> 
> 
> Actually, I can see that your video is 15s long, but awful (and desktop safari) are only showing like, 3 frames looping forever. Did you shorten it, or is something just horribly fucked up?
> — [Zenostein](https://forums.somethingawful.com/showthread.php?noseen=0&threadid=3837546&perpage=40&pagenumber=42#post492621023)

---

> Full rundown:
> I uploaded the video from the Imgur app, then got the link for the mp4 from copying the link within the app and then pasting it into Awful and putting it in as an IMG.
> 
> When I came to my browser (Windows 10, Chrome), the video was not there, so I edited it and took out the IMGs.
> 
> Suddenly, the video appeared and played automatically on desktop, but I only see the hyperlink in mobile.
> 
> That's when I decided to put both in.
> 
> Regarding that 3-second loop: If you refresh, sometimes it will cache the whole thing. It's a cache error.
> — [Wise Fwom Yo Gwave](https://forums.somethingawful.com/showthread.php?noseen=0&threadid=3837546&perpage=40&pagenumber=42#post492622061)

**Comment by geoFlux** [comment]
This has been bugging me for a while. Embedded mp4s work if you use safari browser instead of awful app. It looks like there is a bit of javascript that converts imgur mp4 links so that for example:
```
<a href="https://i.imgur.com/iHzu04n.mp4" target="_blank" rel="nofollow">https://i.imgur.com/iHzu04n.mp4</a>
```
becomes:
```http
<video autoplay loop muted="true" poster="https://i.imgur.com/iHzu04nh.jpg">
    <source src="https://i.imgur.com/iHzu04n.mp4" type="video/mp4">
    <source src="https://i.imgur.com/iHzu04n.webm" type="video/webm">
</video>
```

I took that same javascript and added it to RenderView.js (after converting the jquery stuff to plain javascript). And it works on my machine. Not sure if this is the right approach, but I can make a pull request after I clean it up a bit.

**Comment by realModusOperandi** [comment]
Importing javascript from elsewhere is basically the way we did Twitter embeds so I don't see a problem with that.

**Comment by nolanw** [comment]
I think this was completed by the commits above?
